client/tx/factory.go

```diff
@@ -0,0 +1,164 @@
+		simulateAndExecute: flags.GasFlagVar.Simulate,
+		chainID:            viper.GetString(flags.FlagChainID),
+		memo:               viper.GetString(flags.FlagMemo),
+	}
+
+	f = f.WithFees(viper.GetString(flags.FlagFees))
+	f = f.WithGasPrices(viper.GetString(flags.FlagGasPrices))
+
+	return f
+}
+
+// nolint
+func (f Factory) AccountNumber() uint64              { return f.accountNumber }
+func (f Factory) Sequence() uint64                   { return f.sequence }
+func (f Factory) Gas() uint64                        { return f.gas }
+func (f Factory) GasAdjustment() float64             { return f.gasAdjustment }
+func (f Factory) Keybase() keyring.Keybase           { return f.keybase }
+func (f Factory) ChainID() string                    { return f.chainID }
+func (f Factory) Memo() string                       { return f.memo }
+func (f Factory) Fees() sdk.Coins                    { return f.fees }
+func (f Factory) GasPrices() sdk.DecCoins            { return f.gasPrices }
+func (f Factory) AccountRetriever() AccountRetriever { return f.accountRetriever }
+
+// SimulateAndExecute returns the option to simulate and then execute the transaction
+// using the gas from the simulation results
+func (f Factory) SimulateAndExecute() bool { return f.simulateAndExecute }
+
+// WithTxGenerator returns a copy of the Factory with an updated Generator.
+func (f Factory) WithTxGenerator(g Generator) Factory {
+	f.txGenerator = g
+	return f
+}
+
+// WithAccountRetriever returns a copy of the Factory with an updated AccountRetriever.
+func (f Factory) WithAccountRetriever(ar AccountRetriever) Factory {
+	f.accountRetriever = ar
+	return f
+}
+
+// WithChainID returns a copy of the Factory with an updated chainID.
+func (f Factory) WithChainID(chainID string) Factory {
+	f.chainID = chainID
+	return f
+}
+
+// WithGas returns a copy of the Factory with an updated gas value.
+func (f Factory) WithGas(gas uint64) Factory {
+	f.gas = gas
+	return f
+}
+
+// WithFees returns a copy of the Factory with an updated fee.
+func (f Factory) WithFees(fees string) Factory {
+	parsedFees, err := sdk.ParseCoins(fees)
+	if err != nil {
+		panic(err)
+	}
+
+	f.fees = parsedFees
+	return f
+}
+
+// WithGasPrices returns a copy of the Factory with updated gas prices.
+func (f Factory) WithGasPrices(gasPrices string) Factory {
+	parsedGasPrices, err := sdk.ParseDecCoins(gasPrices)
+	if err != nil {
+		panic(err)
+	}
+
+	f.gasPrices = parsedGasPrices
+	return f
+}
+
+// WithKeybase returns a copy of the Factory with updated Keybase.
+func (f Factory) WithKeybase(keybase keyring.Keybase) Factory {
+	f.keybase = keybase
+	return f
+}
+
+// WithSequence returns a copy of the Factory with an updated sequence number.
+func (f Factory) WithSequence(sequence uint64) Factory {
+	f.sequence = sequence
+	return f
+}
+
+// WithMemo returns a copy of the Factory with an updated memo.
+func (f Factory) WithMemo(memo string) Factory {
+	f.memo = strings.TrimSpace(memo)
```

Comment:
I always thought that it should not be responsibility of `WithMemo()` to sanitise the input string, IMHO this function should just take what the user passes in and let validation happen elsewhere. Plus, it looks inconsistent with the constructor's logic that does not trim spaces: https://github.com/cosmos/cosmos-sdk/pull/5865/files#diff-fd941313ba0722498ead3a7ba98faaecR58

I know that this comes from the legacy `TxBuilder`, but since you are here... :)

Reply:
Sure